cowprotocol/accounting/rewards/mainnet/mainnet_dashboard_query_2510345.sql

```diff
@@ -0,0 +1,222 @@
+        sum(eth_slippage_wei) * 1.0 / pow(10, 18) as slippage
+    from results_per_tx
+    group by solver_address
+),
+
+named_results as (
+    select * from "1541516(end_time='{{end_time}}',cte_name='named_results')"
+),
+-- end SLIPPAGE
+
+-- BEGIN SOLVER REWARDS: https://dune.com/queries/2283297
+batch_rewards_temp as (
+    select  --noqa: ST06
+        rbr.block_deadline,
+        rbr.block_number,  -- Null here means the settlement did not occur.
+        from_hex(rbr.solver) as winning_solver,
+        from_hex(rbr.tx_hash) as tx_hash,
+        -- Unpacking the data
+        cast(cast(rbr.data.winning_score as varchar) as int256) as winning_score,  -- noqa: RF01
+        cast(cast(rbr.data.reference_score as varchar) as int256) as reference_score,  -- noqa: RF01
+        cast(cast(rbr.data.surplus as varchar) as int256) as surplus,  -- noqa: RF01
+        cast(cast(rbr.data.fee as varchar) as int256) as fee,  -- noqa: RF01
+        cast(cast(rbr.data.execution_cost as varchar) as int256) as execution_cost,  -- noqa: RF01
+        cast(cast(rbr.data.uncapped_payment_eth as varchar) as int256) as uncapped_payment_eth,  -- noqa: RF01
+        cast(cast(rbr.data.capped_payment as varchar) as int256) as capped_payment,  -- noqa: RF01
+        transform(rbr.data.participating_solvers, x -> from_hex(x)) as participating_solvers,  -- noqa: RF01
+        cardinality(rbr.data.participating_solvers) as num_participants  -- noqa: RF01
+    from cowswap.raw_batch_rewards as rbr
+    where
+        rbr.block_deadline > (select start_block from block_range)
+        and rbr.block_deadline <= (select end_block from block_range)
+),
+
+batch_rewards as (
+    select
+        block_deadline,
+        block_number,
+        winning_solver,
+        tx_hash,
+        winning_score,
+        reference_score,
+        surplus,
+        fee,
+        execution_cost,
+        capped_payment,
+        participating_solvers,
+        num_participants
+    from batch_rewards_temp
+),
+
+-- AKA Performance Rewards
+primary_rewards as (
+    select
+        winning_solver as solver,
+        cast(sum(capped_payment) as double) as reward_wei
+    from batch_rewards
+    group by winning_solver
+),
+
+fees_and_costs as (
+    select
+        winning_solver as solver,
+        cast(sum(fee) as double) as network_fee_wei,
+        cast(sum(execution_cost) as double) as execution_cost_wei
+    from batch_rewards
+    group by winning_solver
+),
+
+conversion_prices as (
+    select
+        (
+            select avg(
+                case
+                    when price > 10 then 0.26 -- dirty fix for some bogus COW prices Dune reports on July 29, 2024
+                    else price
+                end
+            ) from prices.usd
+            where
+                blockchain = 'ethereum'
+                and contract_address = 0xdef1ca1fb7fbcdc777520aa7f396b4e015f497ab
+                and date(minute) = cast('{{end_time}}' as timestamp) - interval '1' day
```

Comment:
This is fine for now. In the future I would expect this to handle more general accounting periods more gracefully.
A minor improvement would be
```suggestion
                and date(minute) = date(cast('{{end_time}}' as timestamp)) - interval '1' day
```
here and below. (Not super sure though.)

Reply:
Will leave it as is for now

Reply:
A comment regarding the parameter at the top of the file would be helpful. In the sense of
```
--  {{end_time}} - the timestamp for which the analysis should end (exclusively); only choices with with date(end_time) = end_time are allowed
```
Running the query
```sql
select
    date(timestamp '2024-09-16 12:00:00') = timestamp '2024-09-17 00:00:00' - interval '1' day,
    date(timestamp '2024-09-16 12:00:00') = timestamp '2024-09-17 00:00:01' - interval '1' day,
    date(timestamp '2024-09-17 12:00:00') = timestamp '2024-09-17 00:00:01' - interval '1' day
```
gives `true, false, false`. So not choosing `end_time` incorrectly will make this price unavailable.